src/imputation/tmi_imputation.py

```diff
@@ -484,12 +492,78 @@ def run_tmi(
     # TMI Step 5: Calculate headcount and employment totals
     final_df = calculate_totals(expanded_df)
 
-    # add short forms back to dataframe
-    full_df = pd.concat([final_df, shortform_df])
+    TMILogger.info("TMI long form imputation completed.")
+    return final_df, qa_df
+
+
+def run_shortform_tmi(
+    shortform_df: pd.DataFrame,
+    config: Dict[str, Any],
+) -> pd.DataFrame:
+    """Function to run imputation end to end and returns the final
+    dataframe back to the pipeline
+        dataframe back to the pipeline
+    Args:
+        shortform_df (pd.DataFrame): the dataset filtered for long form entries
+        config (Dict): the configuration settings
+    Returns:
+        pd.DataFrame: dataframe with the imputed valued added
+    """
+    TMILogger.info("Starting TMI short form imputation.")
+
+    sf_target_variables = list(config["breakdowns"])
+
+    df = tmi_pre_processing(shortform_df, sf_target_variables)
+
+    mean_dict, qa_df = create_mean_dict(df, sf_target_variables, config)
+
+    # qa_df.set_index("qa_index", drop=True, inplace=True)
+
+    qa_df = qa_df.drop("trim_check", axis=1)
+
+    shortform_tmi_df = apply_tmi(df, sf_target_variables, mean_dict, formtype_short)
+
+    TMILogger.info("TMI imputation completed.")
+    return shortform_tmi_df, qa_df
+
+
+def run_tmi(
+    full_df: pd.DataFrame,
+    sic_mapper: pd.DataFrame,
+    config: Dict[str, Any],
+) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """Function to run imputation end to end and returns the final
+    dataframe back to the pipeline
+        dataframe back to the pipeline
+    Args:
+        full_df (pd.DataFrame): main data
+        sic_mapper (pd.DataFrame): dataframe with sic mapper info
+        config (Dict): the configuration settings
+    Returns:
+        final_df: dataframe with the imputed valued added and counts columns
+        qa_df: qa dataframe
+    """
+    # changing type of Civil or Defence column 200 helps with imputation classes
+    full_df["200"] = full_df["200"].astype("category")
+
+    longform_df = full_df.copy().loc[full_df["formtype"] == formtype_long]
+    shortform_df = full_df.copy().loc[full_df["formtype"] == formtype_short]
```

Comment:
`formtype_long` and `formtype_short` are global variables. For readabililty, maintainability and to uphold the principle of "encapsulation" we should provide that variable as parameter, e.g. `def run_longform_tmi(longform_df, sic_mapper, config, formtype_long)` or perhaps even better (because it will be consistent with other variables, put them in the config, and then you only need to supply the config as an argument. Then it may be accessed with something like `config['global']['long_code']`

Reply:
thing is, we want to split the dataframes before running TMI on the two independently of each other.